File: back-end/go/internal/models/discounted-date.go
package models

import (
    "go.mongodb.org/mongo-driver/bson/primitive"
)

type DiscountedDate struct {
    ID          primitive.ObjectID `json:"id,omitempty" bson:"_id,omitempty"`
    Title       string             `json:"title"`
    Description string             `json:"description"`
    DiscountPct float32            `json:"discount_pct"`
    Date        Date               `json:"date"`
}

File: back-end/nodejs/internal/models/discounted-date.js
var mongoose = require('mongoose');

var schema = new mongoose.Schema({
    _id: mongoose.Schema.Types.ObjectId,
    title: { 
        type: String, 
        required: true 
    },
    description: { 
        type: String, 
        required: true 
    },
    discount_pct: {
        type: Number, 
        required: true 
    },
    date: { 
        type: {
            year: { 
                type: Number, 
                required: true 
            },
            month: { 
                type: Number, 
                required: true 
            },
            day: { 
                type: Number, 
                required: true 
            }
        },
        required: true 
    }
});

var discountedDate = new mongoose.model('DiscountedDate', schema, 'discountedDates');

module.exports = discountedDate;